docs/source/reference/kubernetes/kubernetes-setup.rst

```diff
@@ -13,195 +13,140 @@ Kubernetes Cluster Setup
+.. _kubernetes-gpu-labels:
 
-   .. code-block:: console
+Setting up GPU labels
+^^^^^^^^^^^^^^^^^^^^^
 
-     kubectl get jobs -n kube-system
+.. tip::
 
-4. Verify your kubeconfig (and GPU support, if available) is correctly set up by running :code:`sky check`:
+    If your cluster has the Nvidia GPU Operator installed or you are using GKE or Karpenter, your cluster already has the necessary GPU labels. You can skip this section.
 
-   .. code-block:: console
+To use GPUs with SkyPilot, cluster nodes must be labelled with the GPU type. This informs SkyPilot which GPU types are available on the cluster.
 
-     $ sky check
+Currently supported labels are:
 
+* ``nvidia.com/gpu.product``: automatically created by Nvidia GPU Operator.
+* ``cloud.google.com/gke-accelerator``: used by GKE clusters.
+* ``karpenter.k8s.aws/instance-gpu-name``: used by Karpenter.
+* ``skypilot.co/accelerator``: custom label used by SkyPilot if none of the above are present.
 
-.. _kubernetes-setup-onprem:
+Any one of these labels is sufficient for SkyPilot to detect GPUs on the cluster.
 
-Deploying on on-prem clusters
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. tip::
 
-You can also deploy Kubernetes on your on-prem clusters using off-the-shelf tools,
-such as `kubeadm <https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/>`_,
-`k3s <https://docs.k3s.io/quick-start>`_ or
-`Rancher <https://ranchermanager.docs.rancher.com/v2.5/pages-for-subheaders/kubernetes-clusters-in-rancher-setup>`_.
-Please follow their respective guides to deploy your Kubernetes cluster.
+    To check if your nodes contain the necessary labels, run:
 
-.. _kubernetes-setup-gpusupport:
+    .. code-block:: bash
 
-Setting up GPU support
-~~~~~~~~~~~~~~~~~~~~~~
-If your Kubernetes cluster has Nvidia GPUs, ensure that:
+        output=$(kubectl get nodes --show-labels | awk -F'[, ]' '{for (i=1; i<=NF; i++) if ($i ~ /nvidia.com\/gpu.product=|cloud.google.com\/gke-accelerator=|karpenter.k8s.aws\/instance-gpu-name=|skypilot.co\/accelerator=/) print $i}')
+        if [ -z "$output" ]; then
+          echo "No valid GPU labels found."
+        else
+          echo "GPU Labels found:"
+          echo "$output"
+        fi
 
-1. The Nvidia GPU operator is installed (i.e., ``nvidia.com/gpu`` resource is available on each node) and ``nvidia`` is set as the default runtime for your container engine. See `Nvidia's installation guide <https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html#install-nvidia-gpu-operator>`_ for more details.
-2. Each node in your cluster is labelled with the GPU type. This labelling can be done using `SkyPilot's GPU labelling script <automatic-gpu-labelling_>`_ or by manually adding a label of the format ``skypilot.co/accelerator: <gpu_name>``, where the ``<gpu_name>`` is the lowercase name of the GPU. For example, a node with V100 GPUs must have a label :code:`skypilot.co/accelerator: v100`.
+Manually Labelling Nodes
```

Comment:
Let's have `Automatically Labelling Nodes` before this `Manually Labelling Nodes`, so users do not have to manually labelled all the nodes and then run into the next section for automatic labelling?